### PR TITLE
Improve Code Coverage

### PR DESCRIPTION
**Pull request by alexhersh**
In response to #67

### CODE REVIEW

**Comment by alexhersh** [comment]
Branching off of in-progress work by @Symmetric 
Currently, the only un-accounted for lines involve `calicoctl` calls that are subject to change soon.

File: calico_kubernetes/calico_kubernetes.py [hunk 1]
```diff
@@ -1,23 +1,27 @@
 #!/bin/python
-import json
+from __future__ import print_function
```
Comment (caseydavenport) [inline calico_kubernetes/calico_kubernetes.py]: Unused import?

Comment (alexhersh) [inline calico_kubernetes/calico_kubernetes.py]: nope its used once. print_function turns `print "statement"` into `print("statement")`.
The print_function can be unittested but __builtin__.print can't be easily tested

Comment (alexhersh) [inline calico_kubernetes/calico_kubernetes.py]: Importing this tells the compiler to treat `print` as a function and not a keyword. so `print "statement"` becomes `print("statement")`. This makes it easier to unittest.

Comment (caseydavenport) [inline calico_kubernetes/calico_kubernetes.py]: Right - I'd suggest adding a comment to explain that then, so someone doesn't come along and remove it without having that knowledge.

File: calico_kubernetes/tests/kube_plugin_test.py [hunk 2]
```diff
@@ -108,11 +116,104 @@ def test_delete(self):
...
-            self.assertEqual(namespace, self.plugin.namespace)
-            self.assertEqual(pod_name, self.plugin.pod_name)
-            self.assertEqual(docker_id, self.plugin.docker_id)
-            self.assertEqual(profile_name, self.plugin.profile_name)
-            m_datastore_client.remove_profile(profile_name)
+            assert_equal(namespace, self.plugin.namespace)
+            assert_equal(pod_name, self.plugin.pod_name)
+            assert_equal(docker_id, self.plugin.docker_id)
+            assert_equal(profile_name, self.plugin.profile_name)
+            self.m_datastore_client.remove_profile(profile_name)
```
Comment (Symmetric) [inline calico_kubernetes/tests/kube_plugin_test.py]: Why are you calling this mock function?

File: calico_kubernetes/tests/kube_plugin_test.py [hunk 3]
```diff
@@ -108,11 +116,104 @@ def test_delete(self):
...
+            self.m_datastore_client.remove_profile.side_effect = KeyError
+
+            # Set up args
+            namespace = 'ns'
+            pod_name = 'pod1'
+            docker_id = 123456789101112
+            profile_name = 'ns_pod1_123456789101'
+
+            # Call method under test
+            self.plugin.delete(namespace, pod_name, docker_id)
```
Comment (Symmetric) [inline calico_kubernetes/tests/kube_plugin_test.py]: Much neater -- but I think you should add a comment trailing this function saying something like "We don't need to assert anything here; if WAD we won't raise an exception."

It's a fine optimization to make (which is why I suggested it), but folk reading this code later will wonder if you just forgot to write asserts unless you call out why you omitted them.

File: calico_kubernetes/tests/kube_plugin_test.py [hunk 4]
```diff
@@ -108,11 +116,104 @@ def test_delete(self):
...
+            assert_equal(pod_name, self.plugin.pod_name)
+            assert_equal(docker_id, self.plugin.docker_id)
+            assert_equal(profile_name, self.plugin.profile_name)
+            self.m_datastore_client.remove_profile(profile_name)
+
+    def test_delete_error(self):
+        """Test Pod Deletion Hook Failure"""
+        """
+        If the datastore remove_profile function returns KeyError, the profile is not in the datastore.
+        Issue warning log, but do not fail.
```
Comment (Symmetric) [inline calico_kubernetes/tests/kube_plugin_test.py]: Can you rework this comment please? Characters are cheap, so let's have some full sentences. Something like "For this failure case we expect a warning log, but delete() should not raise an exception". (I found this sentence a bit hard to follow with my "knows nothing about the code" hat on).

**Comment by Symmetric** [comment]
A few more points of review feedback; mostly minor markups, so I think next round should be the last.

File: calico_kubernetes/tests/kube_plugin_test.py (hunk 2 above)
Comment (alexhersh) [inline calico_kubernetes/tests/kube_plugin_test.py]: ooh thats a weird thing that was grandfathered in. I'll fix that.

**Comment by Symmetric** [comment]
LGTM
